global_planner/src/library/cell.cpp

```diff
@@ -70,14 +70,21 @@ Cell Cell::getNeighborFromYaw(double yaw) const {
 }
 
 // Returns the neighbors of the Cell whose risk influences the Cell
-std::vector<Cell> Cell::getFlowNeighbors() const {
-  return std::vector<Cell>{
-      Cell(std::tuple<int, int, int>(xIndex() + 1, yIndex(), zIndex())),
-      Cell(std::tuple<int, int, int>(xIndex() - 1, yIndex(), zIndex())),
-      Cell(std::tuple<int, int, int>(xIndex(), yIndex() + 1, zIndex())),
-      Cell(std::tuple<int, int, int>(xIndex(), yIndex() - 1, zIndex())),
-      Cell(std::tuple<int, int, int>(xIndex(), yIndex(), zIndex() + 1)),
-      Cell(std::tuple<int, int, int>(xIndex(), yIndex(), zIndex() - 1))};
+std::vector<Cell> Cell::getFlowNeighbors(int radius) const {
+  std::vector<Cell> cells;
+  for (int x = -radius; x <= radius; x++) {
+    int y_radius = static_cast<int>(std::ceil(
+        std::sqrt(std::pow(double(radius), 2) - std::pow(double(x), 2))));
+    for (int y = -y_radius; y <= y_radius; y++) {
+      int z_radius = static_cast<int>(std::ceil(
+          std::sqrt(std::pow(double(y_radius), 2) - std::pow(double(y), 2))));
```

Comment:
I think pull out the whole `static_cast<int>(std::ceil(std::sqrt(std::pow(double(radius), 2) - std::pow(double(x), 2))))` into a lambda something like `ceilDistance(radius, x) --> int` will make it a lot nicer to read

Reply:
I have added the lambda in the following commit! This is cool!